source/common/http/codec_client.cc

```diff
@@ -74,6 +84,7 @@ void CodecClient::onEvent(Network::ConnectionEvent event) {
           connected_ ? StreamResetReason::ConnectionTermination
                      : StreamResetReason::ConnectionFailure);
     }
+    disableIdleTimer(true);
```

Comment:
You need to move this before you start closing active requests, around line 80. Also, just do `idle_timer_.reset()`. You don't need the extra parameter to `disableIdleTimer`. Please also add a test for the case where the client is closed with both active requests and active idle timer to https://github.com/envoyproxy/envoy/blob/master/test/common/http/codec_client_test.cc.